src/htm/ntypes/Value.hpp

```diff
@@ -15,121 +17,386 @@
  * along with this program.  If not, see http://www.gnu.org/licenses.
  * --------------------------------------------------------------------- */
 
-/** @file
- * Definitions for the Value class
- *
- * A Value object holds a Scalar or an Array
- * A ValueMap is essentially a map<string, Value>
- * It is used internally in the conversion of YAML strings to C++ objects.
- * The API and implementation are geared towards clarify rather than
- * performance, since it is expected to be used only during network construction.
- */
-
+/***********************************************
+|  Value object - a container for data structures that have been parsed
+|  from a YAML or JSON string.  It can hold any structure that can be
+|  parsed from these but does not implement Alias or reference features.
+|  YAML is a superset of JSON format but we only use the JSON compatible features.
```

Comment:
could we use a JSON parser then? (not that we need, if libyaml works good)

Reply:
> could we use a JSON parser 

Yes, we could.   But I would rather handle both JSON and Yaml.